### PR TITLE
Implement import/export of zipped files

### PR DESCRIPTION
**Pull request by roykho**
Change JSON import/export to zip import/export for easier file transfer and linking.

### CODE REVIEW

File: bookings-helper.php [hunk 1]
```diff
@@ -206,6 +223,89 @@ public function catch_requests() {
...
+		public function open_zip( $filename = '' ) {
+			$zip = new ZipArchive();
+
+			if ( $zip->open( $_FILES['import']['tmp_name'] ) ) {
+				$zip->extractTo( $this->temp_dir );
+				$zip->close();
+
+				$dir = scandir( $this->temp_dir );
+
+				return file_get_contents( $this->temp_dir . '/' . $dir[2] );
```
Comment (bor0) [inline bookings-helper.php]: Should we check if this file exists first?

**Comment by bor0** [comment]
This looks good to me, but I was wondering, should we support both JSON and zip? In case there is no `ZipArchive` for the current installation.

WP core does this:
```
         */
        if ( class_exists( 'ZipArchive', false ) && apply_filters( 'unzip_file_use_ziparchive', true ) ) {
                ...
                return
                ...
        }
        // Fall through to PclZip if ZipArchive is not available, or encountered an error opening the file.
        return _unzip_file_pclzip($file, $to, $needed_dirs);
```

**Comment by roykho** [comment]
@bor0 - Ok another round please. I have reinstated JSON format if ZipArchive library is not installed.

**Comment by bor0** [comment]
@roykho, there is an issue when in ZIP mode and attaching ZIP. However, it works fine when in JSON mode and attaching JSON. See following recording:
untitled12345

Product used from https://github.com/woocommerce/woocommerce-bookings/issues/1312 as an example.

**Pull request merged**